src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs

```diff
@@ -20,36 +22,44 @@ public static ManagedBufferPool Shared
 
         public override OwnedBuffer<byte> Rent(int minimumBufferSize)
         {
-            return new ArrayPoolMemory(minimumBufferSize);
+            return new ArrayPoolBuffer(minimumBufferSize);
         }
 
         protected override void Dispose(bool disposing)
         {
         }
 
-        private class ArrayPoolMemory : ReferenceCountedBuffer<byte>
+        private sealed class ArrayPoolBuffer : OwnedBuffer<byte>
         {
-            public ArrayPoolMemory(int size)
+            byte[] _array;
+            bool _disposed;
+            int _referenceCount;
+            
+            public ArrayPoolBuffer(int size)
             {
                 _array = ArrayPool<byte>.Shared.Rent(size);
             }
 
             public override int Length => _array.Length;
 
+            public override bool IsDisposed => _disposed;
+
+            public override bool IsRetained => _referenceCount > 0;
+
             public override Span<byte> AsSpan(int index, int length)
             {
-                if (IsDisposed) BufferPrimitivesThrowHelper.ThrowObjectDisposedException(nameof(ManagedBufferPool));
+                if (IsDisposed) BufferPrimitivesThrowHelper.ThrowObjectDisposedException(nameof(ArrayPoolBuffer));
                 return new Span<byte>(_array, index, length);
             }
 
             protected override void Dispose(bool disposing)
             {
+                _disposed = true;
```

Comment:
I don't follow. What is the behavior if someone calls Dispose(false) vs Dispose(true)?